common/src/chain/transaction/output/htlc.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: consider removing this in the future when fixed-hash fixes this problem
+#![allow(clippy::non_canonical_clone_impl)]
+
+use crypto::{
+    hash::{self, hash},
+    random::Rng,
+};
+use serialization::{Decode, Encode};
+use zeroize::Zeroize;
+
+use super::{timelock::OutputTimeLock, Destination};
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, serde::Serialize, serde::Deserialize)]
+pub struct HashedTimelockContract {
+    // can be spend either by a specific address that knows the secret
+    pub secret_hash: HtlcSecretHash,
+    pub spend_key: Destination,
+
+    // or by a multisig after timelock expires making it possible to refund
+    pub refund_timelock: OutputTimeLock,
+    pub refund_key: Destination,
+}
+
+#[derive(Zeroize, Clone, Encode, Decode, PartialEq, Eq, Debug)]
```

Comment:
You do not need zeroize for this. The script is short-lived anyway. Zeroize is used for long-term secrets, like private keys.